shopinvader_product_media/tests/test_variant_media.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2024 Camptocamp SA (http://www.camptocamp.com)
+# @author Simone Orsi <simahawk@gmail.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+
+from odoo.tests import SavepointCase
+
+from .common import Mixin
+
+
+class ProductMediaCase(SavepointCase, Mixin):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.template = cls.env.ref("product.product_product_4_product_template")
+        cls.product_a = cls.env.ref("product.product_product_4")
+        cls.product_b = cls.env.ref("product.product_product_4b")
+        cls.product_c = cls.env.ref("product.product_product_4c")
+        cls.media1 = cls._create_storage_media("test-media.pdf")
+        cls.media2 = cls._create_storage_media("test-media.txt")
+        cls.media3 = cls._create_storage_media("test-media.csv")
+        cls.s_product = cls.env["shopinvader.product"].create(
+            {
+                "record_id": cls.template.id,
+                "lang_id": cls.env.ref("base.lang_en").id
+            }
+        )
+        cls.s_variant = cls.env["shopinvader.variant"].create(
+            {
+                "record_id": cls.product_a.id,
+                "shopinvader_product_id": cls.s_product.id
+            }
+        )
+
+    # FIXME
```

Comment:
FIXME! :)
Eg: do not check for the count but for the right media to be in the list.